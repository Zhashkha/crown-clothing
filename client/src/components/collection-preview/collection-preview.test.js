import React from "react";
import { shallow } from "enzyme";

import { CollectionPreview } from "./collection-preview.component";
import CollectionItem from "../collection-item/collection-item.component";

describe("test CollectionPreview", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  let mockRouteName;
  let mockItems;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };
    mockMatch = {
      path: "path"
    };
    mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];
    mockRouteName = "routeName";

    const mockProps = {
      title: "",
      routeName: mockRouteName,
      items: mockItems,
      history: mockHistory,
      match: mockMatch
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push on TitleContainer click", () => {
    wrapper.find("TitleContainer").simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    );
  });

  it("should render only 4 CollectionItem elements", () => {
    expect(wrapper.find(CollectionItem).length).toBe(4);
  });
});
