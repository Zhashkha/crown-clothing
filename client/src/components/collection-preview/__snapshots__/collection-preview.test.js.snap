// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CollectionPreview should match snapshot 1`] = `ShallowWrapper {}`;
